src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs

```diff
@@ -224,7 +225,13 @@ public class StrykerOptions
     /// Instruct Stryker to break execution when at least one test failed on initial run.
     /// </summary>
     public bool BreakOnInitialTestFailure { get; set; }
-    
+
+
+    /// <summary>
+    /// Get/set the mutation id provider
+    /// </summary>
+    public IProvideId MutantIdProvider {get; set;}
```

Comment:
Why is this placed in the options? This doesn't seem the right place to me. Options are settings that can be altered by the user.

Reply:
I am not completely satisfied with this either, but it turns out I had to add it as a parameter to a lot of classes.
I then realized that `Options` is the proper vehicle for this: options is the repository for all customizable/variable behavior.

And we could have a customizable ID generation logic, with two or three options:
1. Technical ID: as a user, I don't care for the ID, I let Stryker decide. This would be as of today
2. Reproducible IDs: as as user, I want IDs to be stable across several runs (on the same code base version). An improved version of previous logic. It could be that each project's IDs would be in specific ranges, such as 0+Id, 1000000+Id, 2000000+Id...
3. Stable IDs: as a user, I want IDs to be stable across code base version. Those should be alphanumeric, using this pattern: _class.member.id_, eg. `MyClass.MyMethed.12`.

This option would clearly be in `Options` and converted to the proper strategy class during project init.

Makes sense?